ui/tests/integration/components/sync/secrets/page/destinations/destination/secrets-test.js

```diff
@@ -84,5 +86,55 @@ module(
       await click(menu.unsync);
       await click(PAGE.confirmButton);
     });
+
+    module('flash messages', function (hooks) {
+      hooks.beforeEach(function () {
+        const flashMessages = this.owner.lookup('service:flash-messages');
+
+        this.flashSuccessSpy = sinon.spy(flashMessages, 'success');
+        this.flashDangerSpy = sinon.spy(flashMessages, 'danger');
+      });
+
+      test('unsync should render flash messages', async function (assert) {
```

Comment:
tbh, this test didn't fail before i merged the fix in, but i couldn't figure out how to reproduce the bug in a test situation given that we need to use mirage as the backend for all things secret-sync. i contemplated adding an assertion in the [adapter test](https://github.com/hashicorp/vault/blob/0529b11571220e6280fd0013358ca23e99d4a47d/ui/tests/unit/adapters/sync/associations-test.js#L101-L109) but ran into the same problem -- i can't make the adapter execute if we're using mirage to intercept HTTP requests 😂

this bug is small enough that i figure it's fine to not have _exact_ test coverage for the situation, but i added these tests regardless because it can't hurt to verify our flash messages are getting triggered at the right time and with the right content.

if y'all have other ideas / suggestions i'm all ears!

Reply:
I agree it's small enough that this is fine! Once thought is doing a test similar to this adapter one. https://github.com/hashicorp/vault/blob/main/ui/tests/unit/adapters/sync/associations-test.js#L139

But stubbing a returned response that it doesn't include the association that's been unsynced so the new conditional setting the id is falsy. I _think_ that'd work? But I'm not certain 😅